mne_bids_pipeline/tests/test_run.py

```diff
@@ -278,3 +282,131 @@ def test_missing_sessions(
+    dst_dir = new_bids_path.root
+    files = [f for f in src_dir.iterdir() if f.is_file()]
+    for _file in files:
+        # in theory we should rewrite `participants.tsv` to remove the `sub-02` line,
+        # but in practice it will just get ignored so we won't bother.
+        shutil.copyfile(src=_file, dst=dst_dir / _file.name)
+    # derivatives (freesurfer files)
+    src_dir = config_obj.bids_root / "derivatives" / "freesurfer" / "subjects"
+    dst_dir = new_bids_path.root / "derivatives" / "freesurfer" / "subjects"
+    dst_dir.mkdir(parents=True)
+    freesurfer_subject_mapping = {"sub-01": "sub-01_ses-a", "sub-02": "sub-01_ses-b"}
+    for walk_root, dirs, files in src_dir.walk():
+        # change "root" so that in later steps of the walk when we're inside a subject's
+        # dir, the "offset" (folders between dst_dir and filename) will be correct
+        new_root = walk_root
+        if "sub-01" in walk_root.parts or "sub-02" in walk_root.parts:
+            new_root = Path(
+                *[freesurfer_subject_mapping.get(p, p) for p in new_root.parts]
+            )
+        offset = new_root.relative_to(src_dir)
+        # the actual subject dirs need their names changed
+        for _dir in dirs:
+            _dir = freesurfer_subject_mapping.get(_dir, _dir)
+            (dst_dir / offset / _dir).mkdir()
+        # for filenames that contain the subject identifier (BEM files, morph maps),
+        # we need to change the filename too, not just parent folder name
+        for _file in files:
+            dst_file = _file
+            for subj in freesurfer_subject_mapping:
+                if subj in dst_file:
+                    dst_file = dst_file.replace(subj, freesurfer_subject_mapping[subj])
+                    break
+            shutil.copyfile(src=walk_root / _file, dst=dst_dir / offset / dst_file)
+    # print_dir_tree(new_bids_path.root)  # for debugging
+    # update config so that `subjects_dir` and `deriv_root` also point to the tempdir
+    extra_config = f"""
+from pathlib import Path
+subjects_dir = "{new_bids_path.root / "derivatives" / "freesurfer" / "subjects"}"
+deriv_root = Path("{new_bids_path.root}") / "derivatives" / "mne-bids-pipeline" / "MNE-funloc-data"
+"""  # noqa E501
+    extra_path = tmp_path / "extra_config.py"
+    extra_path.write_text(extra_config)
+    monkeypatch.setenv("_MNE_BIDS_STUDY_TESTING_EXTRA_CONFIG", str(extra_path))
+    # Run the tests.
+    steps = test_options.get("steps", ())
+    command = ["mne_bids_pipeline", str(config_path), f"--steps={','.join(steps)}"]
+    # hack in the new bids_root
+    command.append(f"--root-dir={new_bids_path.root}")
+    if "--pdb" in sys.argv:
+        command.append("--n_jobs=1")
+    monkeypatch.setenv("_MNE_BIDS_STUDY_TESTING", "true")
+    monkeypatch.setattr(sys, "argv", command)
+    with capsys.disabled():
+        print()
+        main()
+    # check some things that are indicative of different MRIs being used in each session
```

Comment:
this is just the first thing that occurred to me to check, but there's probably a better / simpler / more definitive thing to look at, LMK if you think of something.